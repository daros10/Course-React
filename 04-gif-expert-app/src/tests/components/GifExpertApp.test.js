import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import GifExpertApp from '../../GifExpertApp';

describe('Pruebas al componente GifExpertApp', () => {
  test('Deberia renderizar el componente', () => {
    const wrapper = shallow(<GifExpertApp />);
    expect(wrapper).toMatchSnapshot();
  });

  test('Deberia mostrar una lista de categorias', () => {
    const defaultCategories = ['Rick and Morty', 'Dragon Ball'];
    const wrapper = shallow(
      <GifExpertApp defaultCategories={defaultCategories} />
    );

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('GifGrid').length).toBe(defaultCategories.length);
  });
});
