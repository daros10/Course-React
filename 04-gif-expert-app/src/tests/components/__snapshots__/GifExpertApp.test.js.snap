// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas al componente GifExpertApp Deberia mostrar una lista de categorias 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="Rick and Morty"
      key="Rick and Morty"
    />
    <GifGrid
      category="Dragon Ball"
      key="Dragon Ball"
    />
  </ol>
</Fragment>
`;

exports[`Pruebas al componente GifExpertApp Deberia renderizar el componente 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
